contributors/devel/sig-instrumentation/migration-to-structured-logging.md

```diff
@@ -7,6 +7,33 @@ should be applicable at later stages or for other projects using `klog` logging
 
 [Structured Logging KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
 
+## How to contribute
+
+### About the migration
+
+Migration proceeds in two different ways. First is organized milestone based large scale migration where we are trying to achieve a particular target for a Kubernetes release (for example migrate whole Kubelet code in 1.21). Second is not organized migration done by individual contributors.
+
+Organized migration is done in two state cycle aligned with Kubernetes releases. In stage one we pick a particular migration milestone and create issue that splits the work into smaller chunks (for example migrating just single file) that can be easily picked up by contributors. This way we ensure that multiple people can independently work on common goal and avoid conflicts. After migration is done we can utilize tooling we developed to mark particular directory as migrated to avoid regressions. When we finish particular milestone we move the second stage where we take a break from migration, to analyse results and improve on the process. Adding structured information to logs is very costly and time consuming process, we don't want to make a mistake that would require us to repeat the migration. That's why we reserve time to collect feedback, analyse impact of changes we have done on log volume, performance and PR review process.  
+
+Non-organized migration is done by individual contributors that want particular code section to utilize new features early. Migration of Kubernetes components not yet marked for migration is on best effort bases. This means that we don't guarantee a timely review or that PR will be accepted at all. It's up to component owners to decide if they want to accept those contributions. As this migration is not organized it means that two contributors could start working on migrating same code. Before sending a PR please make sure that there is was already created by someone else.
+
+### Current status
+
+* 1.21 Kubelet was migrated
+* 1.22 We are collecting feedback and making improvements to the migration process.
+
+### Create pr requirements
+
+* It is recommended to create pr according to a package or feature, not one file
+
+* Migrate logs according to log content, reject only rename fuction or modify letter case simply
+
+* Structured Logging migration needs to follow the rules below
+
+## Sending a Structured Logging Migration PR
+
+label with /wg structured-logging and cc @kubernetes/wg-structured-logging-reviews
```

Comment:
```suggestion
## Sending a Structured Logging Migration Pull Request

### Before creating the Pull Request

* Read and understood the. In case of any questions, please contact us on [#wg-structured-logging](https://kubernetes.slack.com/messages/wg-structured-logging) Slack channel.
* Check list of [opened PRs](https://github.com/kubernetes/kubernetes/pulls?q=is%3Aopen+label%3Awg%2Fstructured-logging+is%3Apr) to check if someone is already working on this.

### Getting review

To make sure PR is reviewed in timely manner please add "wg-structured-logging" label and notify our reviewers. To do that please run commends below:

/wg structured-logging 
/cc @kubernetes/wg-structured-logging-reviews

### Why my PR was rejected?

Fact that Kubernetes is organizing migration of to Structured Logging doesn't mean that are accepting all the PRs that migrate logs. We reserve the right to reject the Pull Request in situations listed below:

* Pull request is below minimum quality standards and clearly shows that author hasn't read the guide at all, for example just renamed 'Infof" to 'InfoS'.
* Pull request migrates components which owners decided to not conduct migration at this time. List of those components: kubeadm.

 
```